parser/prism/Translator.cc

```diff
@@ -1014,38 +1014,92 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node, bool preserveCon
         case PM_CALL_NODE: { // A method call like `a.b()` or `a&.b()`
             auto callNode = down_cast<pm_call_node>(node);
 
-            auto loc = location;
-            auto locZeroLen = location.copyWithZeroLength();
-
             auto constantNameString = parser.resolveConstant(callNode->name);
             auto receiver = translate(callNode->receiver);
 
             core::LocOffsets messageLoc;
 
-            // When the message is empty, like `foo.()`, the message location is the same as the call operator location
+            // When the message is empty, like `foo.()`, the message location is the
+            // same as the call operator location
             if (callNode->message_loc.start == nullptr && callNode->message_loc.end == nullptr) {
                 messageLoc = translateLoc(callNode->call_operator_loc);
             } else {
                 messageLoc = translateLoc(callNode->message_loc);
             }
 
+            absl::Span<pm_node_t *> prismArgs;
+            if (auto *prismArgsNode = callNode->arguments) {
+                prismArgs = absl::MakeSpan(prismArgsNode->arguments.nodes, prismArgsNode->arguments.size);
+            }
+
+            // The legacy parser nodes don't include the literal block argument (if any), but the desugar nodes do
+            // include it.
+            core::LocOffsets sendLoc;  // The location of the "send" node, exluding any literal block, if any.
+            core::LocOffsets blockLoc; // The location of just the block node, on its own.
+            core::LocOffsets sendWithBlockLoc = location;
+            location = core::LocOffsets::none(); // Invalidate this to ensure we don't use it again in this path.
```

Comment:
I rewrote this code to not use this ambiguously named "location", so it has to commit to either being the send node loc, or the send+block.